.github/workflows/doctests.yaml

```diff
@@ -0,0 +1,41 @@
+name: Documentation Tests
+
+on:
+  push:
+    branches: [master, v9]
+  pull_request:
+    branches: [master, v9]
+
+permissions:
+  contents: read
+
+jobs:
+  doctests:
+    name: doctests
+    runs-on: ubuntu-latest
+
+    services:
+      redis-stack:
+        image: redis/redis-stack-server:latest
+        options: >-
+          --health-cmd "redis-cli ping" --health-interval 10s --health-timeout 5s --health-retries 5
+        ports:
+          - 6379:6379
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [ 1.18.x, 1.19.x, 1.20.x ]
```

Comment:
IMHO for the purposes of the doctests, you can rely on the oldest. 1.18 -> 1.20 is fine for these. I don't think we need that degree of coverage, as it's duplication.